src/components/modules/epics/ContributionsEpic.tsx

```diff
@@ -188,16 +196,22 @@ const EpicBody: React.FC<BodyProps> = ({
     );
 };
 
-export const ContributionsEpic: React.FC<EpicProps> = ({
+export const getContributionsEpicComponent: (
+    optOutVariant: EpicArticleCountOptOutTestVariants,
+) => React.FC<EpicProps> = optOutVariant => ({
     variant,
     tracking,
     countryCode,
     numArticles,
     onReminderOpen,
     email,
     submitComponentEvent,
+    openCmp,
+    hasConsentForArticleCount,
 }: EpicProps) => {
     const [isReminderActive, setIsReminderActive] = useState(false);
+    const [hasOptedOut, setHasOptedOut] = useState(hasArticleCountOptOutCookie());
+
```

Comment:
It might be quite nice to pull some of this logic out into it's own custom hook! That makes it a) reusable (not that we really need it to be), but also b) slightly less code in the epic which is always nice. Custom hooks are just functions that you write that make use of the base hooks (e.g `useState`). 

We could create a custom hook, say `useArticleCountOptOut` that looks like: 

```ts
interface ArticleCountOptOut {
    hasOptedOut: boolean;
    onArticleCountOptOut: () => void;
    onArticleCountOptIn: () => void;
}

function useArticleCountOptOut(): ArticleCountOptOut {
    const [hasOptedOut, setHasOptedOut] = useState(hasArticleCountOptOutCookie());

    function onArticleCountOptOut() {
        setHasOptedOut(true);
        addArticleCountOptOutCookie();
        removeArticleCountFromLocalStorage();
    }

    function onArticleCountOptIn() {
        setHasOptedOut(false);
        removeArticleCountOptOutCookie();
    }

    return { hasOptedOut, onArticleCountOptOut, onArticleCountOptIn };
}
```

and then use this inside the epic component:

```ts
    const { hasOptedOut, onArticleCountOptIn, onArticleCountOptOut } = useArticleCountOptOut();
```

It's maybe in a slightly strange place, but we have a `hooks` folder that we could add this to (i.e in `useArticleCountOptOut.ts`)